### ISSUE TITLE
Make saver.restore work with tf.keras.layers when using tf.estimator to save model checkpoints

### ISSUE THREAD

**Issue by capilano** [tensorflow/tensorflow#30233]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using):1.14
- Are you willing to contribute it (Yes/No):

**Describe the feature and the current behavior/state.**
 How do I  convert a saved tensorflow estimator model to a Keras model? (either by using the check pointed weights or the saved model) It is possible to convert a compiled Keras model to an estimator instance but it would be great if there is a clean way to convert an estimator saved model back to a keras model without having to manually copy weights. 

**Will this change the current api? How?**
Not sure

**Who will benefit with this feature?**
Will help folks using tf estimators to train and save model checkpoints and restore them directly when a tf.keras model is instantiated.

**Any Other info.**

**Comment by capilano** [comment]
The tf.keras layers do not follow the same naming convention used in the estimator model checkpoint files. So, it becomes a mess to restore models. I am not sure if this is a legitimate issue or if I missing something while restoring

**Comment by ravikyram** [comment]
Can you please elaborate about the issue & the context.Will it be possible to provide related code.Thanks!

**Comment by capilano** [comment]
Have updated the form

**Comment by k-w-w** [comment]
There aren't any plans to update `tf.train.Saver`, given that Keras checkpoints use a somewhat different saving format (`tf.train.Checkpoint`). I've created a gist demonstrating how to save/load the weights in Estimator and Keras. The main change to the usual Estimator code is the use of `tf.train.Checkpoint`.

https://gist.github.com/k-w-w/01c58789c2baf2a350c28e5a89566420

**Comment by capilano** [comment]
@k-w-w Thanks, I cannot remember exactly now but I think I ran into issues when creating a network using tf.keras.layers without using a keras model, I just replaced tf.layers with tf.keras.layers and something broke somewhere but I cannot precisely remember where.

**Comment by k-w-w** [comment]
I believe there are some minor differences between tf.layers and tf.keras.layers which may affect the resulting checkpoint. If you can reproduce the steps, that would be extremely helpful. (and don't forget to re-open the bug)

**Comment by capilano** [comment]
@k-w-w I just looked at the gist you made. I did not explicitly create tf.train.Checkpoint and pass it to the scaffold class like you did and since the estimator(I think) uses tf.train.Saver under the hood, it expects the variable names to be the same while restoring. One other difference I noticed(not directly related to this) is that tf.keras.layers will rename layers even if a name is provided to the layer if a layer with the same name exists within a session, and also it does not reuse layers even if a scope is provided.(I forgot about this back then)
So, since all this is expected behavior, I do not think the issue needs to be reopened. Thanks
